django/contrib/admindocs/views.py

```diff
@@ -167,7 +167,9 @@ class ViewDetailView(BaseAdminDocsView):
     @staticmethod
     def _get_view_func(view):
         urlconf = get_urlconf()
-        if get_resolver(urlconf)._is_callback(view):
+        urlresolver = get_resolver(urlconf)
+        urlresolver._populated = False
```

Comment:
This doesn't quite seem right, combined with the call to `register_callback()` in `_is_callback()`.

I would have expected to have the ability for `django.contrib.admindocs` to register it's callback - by doing so in a `ready()` method on `django.contrib.admindocs.apps.AdminDocsConfig`[^1]  - with `django.urls` which would provide the interface. It would then call each callback that was registered with the `urlconf` and `urlpattern` at the places that were removed in `URLResolver._populate()`.

[^1]: See `django.contrib.admin.apps` for how the admin registers with the checks framework.